### PR TITLE
Add composite provider to support multiple network providers

### PR DESCRIPTION
**Pull request by lujiajing1126**
### Ⅰ. Describe what this PR does

Currently, only a single trafficRouting network provider is supported even multiple ones are defined.

To accomplish this, a composite provider is needed. A composite provider is just an array of `NetworkProvider`.

### CODE REVIEW

**Comment by zmberg** [comment]
@lujiajing1126 There is no problem with the code logic and implementation. We didn't implement it before mainly because we didn't think through the scenarios, so would it be convenient for you to participate in a community meeting? Time: 7.11. Alternatively, you can join the dingding group and we'll talk further!

**Comment by lujiajing1126** [comment]
> @lujiajing1126 There is no problem with the code logic and implementation. We didn't implement it before mainly because we didn't think through the scenarios, so would it be convenient for you to participate in a community meeting? Time: 7.11. Alternatively, you can join the dingding group and we'll talk further!

Yes, sure. I've already in the Dingtalk group and will attend the community meeting on 11, Jul.

File: pkg/trafficrouting/network/composite.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+func (c CompositeController) Initialize(ctx context.Context) error {
+	var err error
+	for _, provider := range c {
+		err = multierr.Append(err, provider.Initialize(ctx))
+	}
+	return err
+}
+
+func (c CompositeController) EnsureRoutes(ctx context.Context, strategy *v1beta1.TrafficRoutingStrategy) (bool, error) {
```
Comment (lujiajing1126) [inline pkg/trafficrouting/network/composite.go:42]: I suppose the only point in dispute is how to handle error from a single networkProvider.

Can you have a check? @zmberg @furykerry

File: pkg/trafficrouting/network/composite.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+func (c CompositeController) Initialize(ctx context.Context) error {
+	for _, provider := range c {
+		if err := provider.Initialize(ctx); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c CompositeController) EnsureRoutes(ctx context.Context, strategy *v1beta1.TrafficRoutingStrategy) (bool, error) {
```
Comment (lujiajing1126) [inline pkg/trafficrouting/network/composite.go:43]: Follow the logic of `CustomNetworkProvider` here.

Gentle Ping @myname4423 @zmberg for review

**Comment by lujiajing1126** [comment]
Test cases have been moved to v1beta1 file and a workflow has been added. @myname4423

**Pull request merged**